examples/ct/ct_handlers.go

```diff
@@ -19,48 +29,76 @@ const (
 )
 
 const (
-	jsonMapKeyCertificates string = "certificates"
+	// HTTP content type header
+	contentTypeHeader      string     = "Content-Type"
+	// MIME content type for JSON
+	contentTypeJSON        string     = "application/json"
+	// The name of the JSON response map key in get-roots responses
+	jsonMapKeyCertificates string     = "certificates"
+	// Logging level for debug verbose logs
+	logVerboseLevel        glog.Level = 2
 )
 
 // CTRequestHandlers provides HTTP handler functions for CT V1 as defined in RFC 6962
 // and functionality to translate CT client requests into forms that can be served by a
 // log backend RPC service.
 type CTRequestHandlers struct {
-	trustedRoots *PEMCertPool
-	rpcClient    trillian.TrillianLogClient
+	logID         int64
+	trustedRoots  *PEMCertPool
+	rpcClient     trillian.TrillianLogClient
+	logKeyManager crypto.KeyManager
+	rpcDeadline   time.Duration
+	timeSource    util.TimeSource
 }
 
 // NewCTRequestHandlers creates a new instance of CTRequestHandlers. They must still
 // be registered by calling RegisterCTHandlers()
-func NewCTRequestHandlers(trustedRoots *PEMCertPool, rpcClient trillian.TrillianLogClient) *CTRequestHandlers {
-	return &CTRequestHandlers{trustedRoots, rpcClient}
+func NewCTRequestHandlers(logID int64, trustedRoots *PEMCertPool, rpcClient trillian.TrillianLogClient, km crypto.KeyManager, rpcDeadline time.Duration, timeSource util.TimeSource) *CTRequestHandlers {
+	return &CTRequestHandlers{logID, trustedRoots, rpcClient, km, rpcDeadline, timeSource}
 }
 
 func pathFor(req string) string {
 	return ctV1BasePath + req
 }
 
+// addChainRequest is a struct for parsing JSON add-chain requests. See RFC 6962 Sections 4.1 and 4.2
 type addChainRequest struct {
 	Chain []string
 }
 
+// addChainResponse is a struct for marshalling add-chain responses. See RFC 6962 Sections 4.1 and 4.2
+type addChainResponse struct {
+	SctVersion int    `json:sct_version`
+	ID         string `json:id`
+	Timestamp  uint64 `json:timestamp`
+	Extensions string `json:extensions`
+	Signature  string `json:signature`
+}
+
 func parseBodyAsJSONChain(w http.ResponseWriter, r *http.Request) (addChainRequest, error) {
 	body, err := ioutil.ReadAll(r.Body)
 
 	if err != nil {
-		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		if glog.V(logVerboseLevel) {
+			glog.Info("Failed to read request body")
+		}
+		sendHttpError(w, http.StatusBadRequest, err)
 		return addChainRequest{}, err
 	}
 
 	var req addChainRequest
 	if err := json.Unmarshal(body, &req); err != nil {
-		http.Error(w, http.StatusText(http.StatusBadRequest)+": "+err.Error(), http.StatusBadRequest)
+		glog.V(logVerboseLevel).Info("Failed to read request body")
```

Comment:
Could you make this log message distinct from the one on line 83 please? "Failed to decode request body" would be fine. Is there a reason you don't log `err`?